### PR TITLE
Webhook documentation

### PR DESCRIPTION
**Pull request by alanhartless**
Ports https://developer.mautic.org/#webhooks6

<a href="https://gitpod.io/#https://github.com/mautic/developer-documentation-new/pull/14"></a>

### CODE REVIEW

File: docs/webhooks/events/lead_company_change.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
+Contact Company subscription change event
+------------------------------------------
+
+Triggered when Mautic adds or removes a Company to/from a Company.
```
Comment (RCheesley) [inline docs/webhooks/events/lead_company_change.rst:4]:
```suggestion
Triggered when Mautic adds or removes a Contact to/from a Company.
```

I think this should be when a contact is added/removed.

File: docs/webhooks/events/lead_post_delete.rst [hunk 2]
```diff
@@ -0,0 +1,30 @@
+Contact deleted event
+----------------------------
+Triggered when Mautic deletes Contact.
```
Comment (RCheesley) [inline docs/webhooks/events/lead_post_delete.rst:3]:
```suggestion
Triggered when Mautic deletes a Contact.
```

File: docs/webhooks/events/lead_post_save_new.rst [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+      - array
+      - Currently not populated for Webhooks.
+    * - ``utmtags``
+      - array
+      - Currently not populated for Webhooks.
+    * - ``stage``
+      - object
+      - Currently not fully populated for Webhooks.
+
+.. _Custom field properties:
```
Comment (RCheesley) [inline docs/webhooks/events/lead_post_save_new.rst:105]:
```suggestion
.. _Custom Field properties:
```

I think both should be capitalised as it's the name of a feature of Mautic.

File: docs/webhooks/events/lead_post_save_new.rst [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+
+Custom field properties
```
Comment (RCheesley) [inline docs/webhooks/events/lead_post_save_new.rst:107]:
```suggestion
Custom Field properties
```

File: docs/webhooks/events/lead_post_save_new.rst [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+
+.. list-table::
+    :header-rows: 1
+
+    * - Key
+      - Type
+      - Description
+    * - ``id``
+      - int
+      - ID of the channel subscription entry.
```
Comment (RCheesley) [inline docs/webhooks/events/lead_post_save_new.rst:244]:
```suggestion
      - ID of the Channel subscription entry.
```

File: docs/webhooks/events/lead_post_save_new.rst [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+    * - ``reason``
+      - int
+      - Reason code for the unsubscription. ``1`` is unsubscribed by the Contact, ``2`` is bounced, and ``3`` is manually marked as unsubscribed by the Marketer.
+    * - ``channel``
+      - string
+      - Channel the Contact unsubscribed from. Examples are ``email`` and ``sms``.
+    * - ``channelId``
+      - int|null
+      - ID of the specific channel entity the Contact clicked to unsubscribe from.
```
Comment (RCheesley) [inline docs/webhooks/events/lead_post_save_new.rst:253]:
```suggestion
      - ID of the specific Channel entity the Contact clicked to unsubscribe from.
```

File: docs/webhooks/getting_started.rst [hunk 7]
```diff
@@ -1,4 +1,110 @@
 Getting started with webhooks
 ==========================================================
 
+Webhooks are a universal way how to send data about Contacts and their activity to a third party in either real-time, as the change/activity happens, or queued to be sent in batches through background cron jobs.
```
Comment (RCheesley) [inline docs/webhooks/getting_started.rst:4]:
```suggestion
Webhooks are a universal way to send data about Contacts and their activity to a third party in either real-time - as the change/activity happens - or queued - to be sent in batches through background cron jobs.
```

File: docs/webhooks/getting_started.rst [hunk 8]
```diff
@@ -1,4 +1,110 @@
...
 
+The structure of Webhook payloads are like the following. ``WebhookEventType`` would be the event's actual type and `WebhookEventPayload` will contain the event's data.
```
Comment (RCheesley) [inline docs/webhooks/getting_started.rst:6]:
```suggestion
The structure of Webhook payloads are as follows. ``WebhookEventType`` would be the event's actual type and `WebhookEventPayload` will contain the event's data.
```

**Review by RCheesley — CHANGES_REQUESTED** [review]
A few minor changes mostly to grammar :)

Thanks for writing this up, great job! 🚀

**Comment by alanhartless** [comment]
@dennisameling @RCheesley I think this is an example of where we may not want to convert Sphinx warnings into errors. 

This is failing to build because of 
```
Warning, treated as error:
/home/runner/work/developer-documentation-new/developer-documentation-new/docs/webhooks/events/company_post_save.rst:8:duplicate label event type, other instance in /home/runner/work/developer-documentation-new/developer-documentation-new/docs/webhooks/events/company_post_delete.rst
```
These were defined this way on purpose because I didn't want to uniquely name `Event type` and `Event properties` for every Webhook as they'll end up with really long titles for some to ensure unique across contact or company related ones. These are also defined as document paragraphs (`""""""""""""""""""`) which are converted to fragments instead of paths so I think this should really be considered a bug in the linter.

**Comment by RCheesley** [comment]
If it's a bug in the linter, then we'd make an upstream issue report I guess to Vale?

If the error is something we can safely skip then let's get it merged.

**Pull request merged**